src/plugins/dashboard/public/dashboard_container/embeddable/create/performance/query_performance_tracking.test.ts

```diff
@@ -0,0 +1,240 @@
+      expect(dashboard.creationEndTime).toBeDefined();
+    });
+  });
+
+  it('Reports a metric event when all panels with phase event reporting have rendered', async () => {
+    const children = {
+      panel1: {
+        phase$: new BehaviorSubject<PhaseEvent>({ status: 'loading', id: '', timeToEvent: 0 }),
+      },
+      panel2: {
+        phase$: new BehaviorSubject<PhaseEvent>({ status: 'loading', id: '', timeToEvent: 0 }),
+      },
+    };
+    const { dashboard } = mockDashboard(children);
+    startQueryPerformanceTracking(dashboard);
+
+    expect(mockMetricEvent).not.toHaveBeenCalled();
+    setChildrenStatus(children, 'rendered');
+
+    expect(mockMetricEvent).toHaveBeenCalledWith(
+      expect.objectContaining({
+        eventName: 'dashboard_loaded',
+        key1: 'time_to_data',
+        value1: expect.any(Number),
+        key2: 'num_of_panels',
+        value2: 2,
+        key4: 'load_type',
+        value4: 1, // dashboard first load
+      })
+    );
+  });
+
+  it('ignores panels that do not publish phase events', async () => {
+    const children = {
+      panel1: {
+        phase$: new BehaviorSubject<PhaseEvent>({ status: 'loading', id: '', timeToEvent: 0 }),
+      },
+      panel2: {
+        phase$: new BehaviorSubject<PhaseEvent>({ status: 'loading', id: '', timeToEvent: 0 }),
+      },
+      panel3: { wow: 'wow' },
+      panel4: { wow: 'wow' },
+    };
+    const { dashboard } = mockDashboard(children);
+    startQueryPerformanceTracking(dashboard);
+    setChildrenStatus(children, 'rendered');
+
+    expect(mockMetricEvent).toHaveBeenCalledWith(
+      expect.objectContaining({
+        eventName: 'dashboard_loaded',
+        key1: 'time_to_data',
+        value1: expect.any(Number),
+        key2: 'num_of_panels',
+        value2: 4,
+        key4: 'load_type',
+        value4: 1, // dashboard first load
+      })
+    );
+  });
+
+  it('reports subsequent loads', async () => {
+    const children = {
+      panel1: {
+        phase$: new BehaviorSubject<PhaseEvent>({ status: 'loading', id: '', timeToEvent: 0 }),
+      },
+      panel2: {
+        phase$: new BehaviorSubject<PhaseEvent>({ status: 'loading', id: '', timeToEvent: 0 }),
+      },
+      panel3: { wow: 'wow' },
+      panel4: { wow: 'wow' },
+    };
+    const { dashboard } = mockDashboard(children);
+    startQueryPerformanceTracking(dashboard);
+    setChildrenStatus(children, 'rendered');
+
+    await waitFor(() => {
+      expect(mockMetricEvent).toHaveBeenCalledWith(
+        expect.objectContaining({
+          value4: 1, // dashboard first load
+        })
+      );
+    });
+
+    setChildrenStatus(children, 'loading');
+    await new Promise((r) => setTimeout(r, 1));
+    setChildrenStatus(children, 'rendered');
+
+    expect(mockMetricEvent).toHaveBeenCalledWith(
+      expect.objectContaining({
+        value4: 2, // dashboard subsequent load
+      })
+    );
+  });
+
+  it('reports Dashboard first load', () => {});
```

Comment:
Should this test be filled out? :)

Reply:
Whoops, meant to remove that. It's captured in the test above - line 173